### PR TITLE
Add support for displaying in Elixir format

### LINKED ISSUE #94: Elixir syntax formatting
erlyberly should allow terms and functions to be displayed in elixir style e.g. `:my_atom` not `my_atom`.

erlang and LFE is already supported, these can be used as a guide as to how to include this for elixir.

This requires an elixir specific formatter class, the erlang one can be used as a template: https://github.com/andytill/erlyberly/blob/master/src/main/java/erlyberly/format/ErlangFormatter.java

An entry in the preferences FXML file: https://github.com/andytill/erlyberly/blob/master/src/main/resources/erlyberly/preferences.fxml#L36

And support in the PreferencesView class: https://github.com/andytill/erlyberly/blob/master/src/main/java/erlyberly/PreferencesView.java#L80

### PR DESCRIPTION
**Pull request by aboroska**
Elixir support arrived! :) At least the first version.

The following is displayed in the trace window:

screen shot 2017-01-11 at 22 44 07

while tracing Application.put_env/3 when running the following in IEx:
---------------------------------------------------------------------

iex(my@localhost)80> Application.put_env :an, :elixir_string, "Elixir string"
:ok
iex(my@localhost)81> Application.put_env :an, :erlang_string, 'erlang string'
:ok
iex(my@localhost)82> Application.put_env :an, :atom, :hello_atom
:ok
iex(my@localhost)83> Application.put_env :a, :list, [1,2,3]
:ok
iex(my@localhost)84> Application.put_env :a, :binary, <<1,2,3>>
:ok
iex(my@localhost)85> Application.put_env :a, :tuple, {1,2}
:ok
iex(my@localhost)86> Application.put_env :a, :map, %{foo: "bar"}
:ok
iex(my@localhost)87> Application.put_env :a, :map, %{"hello" => "world", a: 1, b: 2}
:ok
iex(my@localhost)88> Application.put_env :a, :map, %{"hello" => %{inline: "map"}, a: 1, b: 2}
:ok 

Fixes #94.

### CODE REVIEW

**Comment by aboroska** [comment]
Shall I stash the two commits?

**Comment by ruanpienaar** [comment]
Stash your trash! :)

File: src/main/java/erlyberly/format/Formatting.java [hunk 1]
```diff
@@ -108,4 +108,25 @@ public static void binaryToString(OtpErlangBinary bin, String sep, StringBuilder
...
         return b > 31 && b < 127;
     }
+
+    public static boolean isDisplayableString(OtpErlangBinary bin) {
+        int expected;
+        byte[] bytes = bin.binaryValue();
+        for (int i = 0; i < bytes.length; i++) {
+            int ch = bytes[i];
+            if (isDisplayableChar(ch)) continue;
+            else if ((ch & 0b11100000) == 0b11000000) expected = 1;
```
Comment (andytill) [inline src/main/java/erlyberly/format/Formatting.java]: Is this utf-8 support? Could you add a comment describing what it is doing?

**Comment by andytill** [comment]
-[ ] Atoms like this in erlang 'Hi There' get printed in the same way when using elixir format, which makes it look like a char list.

**Comment by andytill** [comment]
Great work! I'm very happy about this going in.

Don't worry about squashing the commits, I will do that on merge, or not if you prefer.

File: src/main/java/erlyberly/format/Formatting.java (hunk 1 above)
Comment (aboroska) [inline src/main/java/erlyberly/format/Formatting.java]: Comment added.

**Comment by aboroska** [comment]
Fixed the atoms.

**Pull request merged**